compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/InitMethodGen.java

```diff
@@ -254,24 +388,58 @@ private Name getNextVarId() {
         return new Name(varIdPrefix + nextVarId);
     }
 
-    private BIRNode.BIRBasicBlock addCheckedInvocation(BIRNode.BIRFunction func, PackageID modId, String initFuncName,
-                                                       BIROperand retVar, BIROperand boolRef) {
+    private BIRNode.BIRBasicBlock addCheckedInvocationForExitCall(BIRNode.BIRFunction func, BIROperand retVar,
+                                                                  BIROperand boolRef, String typeOwnerClass) {
+        BIRNode.BIRBasicBlock lastBB = func.basicBlocks.get(func.basicBlocks.size() - 1);
+        BIRNode.BIRBasicBlock nextBB = addAndGetNextBasicBlock(func);
+
+        lastBB.terminator = getExitMethodCall(nextBB, typeOwnerClass);
+
+        BIRNonTerminator.TypeTest typeTest = new BIRNonTerminator.TypeTest(null, symbolTable.errorType, boolRef,
+                retVar);
+        nextBB.instructions.add(typeTest);
+
+        BIRNode.BIRBasicBlock trueBB = addAndGetNextBasicBlock(func);
+        BIRNode.BIRBasicBlock retBB = addAndGetNextBasicBlock(func);
+        retBB.terminator = new BIRTerminator.Return(null);
+        trueBB.terminator = new BIRTerminator.GOTO(null, retBB);
+
+        BIRNode.BIRBasicBlock falseBB = addAndGetNextBasicBlock(func);
+        nextBB.terminator = new BIRTerminator.Branch(null, boolRef, trueBB, falseBB);
+        return falseBB;
+    }
+
+    private static JInternalCall getExitMethodCall(BIRNode.BIRBasicBlock nextBB, String typeOwnerClass) {
+        JInternalCall jiMethodCall = new JInternalCall(null);
+        jiMethodCall.args =  new ArrayList<>();
+        jiMethodCall.varArgExist = false;
+        jiMethodCall.jClassName = typeOwnerClass;
+        jiMethodCall.jMethodVMSig = GRACEFUL_EXIT_METHOD;
+        jiMethodCall.name = "graceFulExit";
```

Comment:
Shall we change all the occurrences of the word `graceFul` to `graceful`
```suggestion
        jiMethodCall.name = "gracefulExit";
```